src/db-api.js

```diff
@@ -68,47 +68,141 @@ const transactionsHistoryForAddresses = (db: Pool) => async (
-      AND tx_state = $2
-      ORDER BY time DESC`,
-    values: [addresses, 'Successful'],
-  })
-
 /**
 * Queries TXS table looking for a successful transaction with a given hash
 * @param {Db Object} db
-* @param {*} tx
+* @param {Transaction} tx
 */
 const txSummary = (db: Pool) => async (tx: string): Promise<ResultSet> =>
   db.query({
     text: 'SELECT * FROM "txs" WHERE hash = $1 AND tx_state = $2',
     values: [tx, 'Successful'],
   })
 
+/**
+* Queries TX table looking for tx by its hash
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getTx = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT id, block, hash::text FROM "tx" WHERE hash = $1',
+    values: [tx],
+  })
+
+/**
+* Queries BLOCK table looking for block with a given id
+* @param {Db Object} db
+* @param {Block} blockId
+*/
+const getBlockById = (db: Pool) => async (blockId: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT time, block_no, hash::text FROM block WHERE id = $1',
+    values: [blockId],
+  })
+
+/**
+* Quries TX* tables to get txInputs for a given transaction
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getTxInputs = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
+    text: `SELECT
+      tx_out.address, tx_out.value
+      FROM tx_out
+      INNER JOIN tx ON tx.id = tx_out.tx_id
+      INNER JOIN tx_in ON tx_in.tx_out_id = tx_out.tx_id AND tx_in.tx_out_index = tx_out.index
+      WHERE tx_in.tx_in_id = $1`,
+    values: [tx],
+  })
+
+/**
+* Queries TX, BLOCK, TX_OUT tables to acquire transactions that went into given addresses
+* @param {Db Object} db
+* @param {Array<Address>} addresses
+*/
+const getInwardTransactions = (db: Pool) => async (addresses: Array<string>): Promise<ResultSet> =>
+  db.query({
+    text: `SELECT
+      tx.id, tx.hash::text, block.time
+      FROM block 
+      INNER JOIN tx ON block.id = tx.block 
+      INNER JOIN tx_out ON tx.id = tx_out.tx_id
+      WHERE tx_out.address = ANY($1)`,
+    values: [addresses],
+  })
+
+/**
+* Queries TX, BLOCK, TX_OUT tables to acquire transactions that went into given addresses
+* @param {Db Object} db
+* @param {Array<Address>} addresses
+*/
+const getOutwardTransactions = (db: Pool) => async (addresses: Array<string>): Promise<ResultSet> =>
+  db.query({
+    text: `SELECT DISTINCT 
+      tx.id, tx.hash::text, block.time
+      FROM block 
+      INNER JOIN tx ON block.id = tx.block 
+      INNER JOIN tx_in ON tx.id = tx_in.tx_in_id 
+      INNER JOIN tx_out ON (tx_in.tx_out_id = tx_out.tx_id) AND (tx_in.tx_out_index = tx_out.index)
+      WHERE tx_out.address = ANY($1)`,
+    values: [addresses],
+  })
+
+/**
+* Queries TX* tables to acquire distinct tx inputs for given transactions
+* @param {Db Object} db
+* @param {Array<Transaction>} txs
+*/
+const getDistinctTxInputs = (db: Pool) => async (txs: Array<string>): Promise<ResultSet> =>
```

Comment:
I think for the function user it's implicit that the inputs are deduplicated so I would name the function just `getTxInputs`